akd/src/storage/memory.rs

```diff
@@ -30,15 +31,42 @@ type UserValueMap = HashMap<Epoch, ValueState>;
 pub struct AsyncInMemoryDatabase {
     db: Arc<DashMap<Vec<u8>, DbRecord>>,
     user_info: Arc<DashMap<Vec<u8>, UserValueMap>>,
+    /// This flag is used to determine whether the database will automatically
+    /// (and aggressively) remove entries corresponding to left and right
+    /// children of a tree node when the node is inserted. The purpose behind this
+    /// is to reduce the size of the in-memory database by culling the child enries
+    /// once the parent node's hash has been calculated. The primary use case for this
+    /// is to improve auditing memory usage and time (since during auditing, we no longer
+    /// care about the child node hashes once its parent has been computed). Note that this
+    /// technique takes advantage of the way batch insertion of nodes into the tree works,
+    /// since we always process all of the children of a particular subtree before processing
+    /// the root of that subtree.
+    remove_child_nodes_on_insertion: bool,
```

Comment:
I'm generally not the biggest fan of using a boolean to differentiate behavior, as I'd like to use something like a different type to reflect that the in-memory store we're using doesn't store everything, but I think that's a bit more of a rework than what we have now. 

As such, I think what we have here is sufficient since we need to influence how inner workings of `batch_set` and using something like a newtype isn't necessarily going to make that easy given that we're not calling something before or after existing functionality. Additionally, you've commented this really well so it's pretty clear + the associated function to instantiate is super clear 👍